### ISSUE TITLE
No multiple instances

### ISSUE THREAD

**Issue by DocMAX** [Martchus/syncthingtray#142]
Can syncthingtray just do nothing if it's allready running?
Currently it adds a new tray icon which makes no sense.

**Comment by DocMAX** [comment]
My usecase is that i sometimes start a second x11 session and syncthingtray is in autostart. After this i have 2 tray icons on my first sessions tray. I can workaround that with a script, to check if syncthingtray is allready running. Yes it's the Qt GUI.

**Comment by Martchus** [comment]
Ok, that use-case makes sense. How do you autostart Syncthing Tray? Using the option in Syncthing Tray's settings? If that's so I suppose it would indeed make sense to add an option to avoid starting another instance and use that for the autostart. And if you somehow configure the autostart manually you can of course also simply pass that option then.

The only concern is what would happen if one logs in as a different user that also has the autostart enabled (but that might be buggy as-is anyways).

**Comment by Martchus** [comment]
It is implemented on master, see commits mentioned by GitHub. Note that it doesn't work when you're using a different user because then the whole single instance behavior doesn't work. (I've just fixed it so at least each user can have his own independent single instance.)

**Comment by gabeweb** [comment]
Hi there,

Honestly, this app is too difficult to use on Windows. I switched from SyncTrayzor and I'm having trouble getting Syncthing Tray to run two instances simultaneously. I use Syncthing on my laptop to sync with my USB drive on the same device, but it's impossible because both instances close each other and see the same folders (the folders on my laptop).

I've tried copying my current SyncTrayzor configuration into Syncthing Tray, but nothing happens.

However, syncing one of the instances with my mobile phone works fine. The only difficult thing is using two instances on the same device.

**Comment by Martchus** [comment]
Checkout the help text via `--help`. You'll find an option to spawn multiple instances. This should probably be mentioned in the README as well.

Parallel running instances will still use the same config by default. You can place the config file right next to the exe for a portable installation that uses its own config, e.g. on the USB drive. The README contains a section explaining where the config is located.

**Comment by Martchus** [comment]
Note that you most likely want to use a different tool like rsync to transfer files to the USB drive. Syncthing is not made for this use case and using it nevertheless seems quite suboptimal.

**Comment by gabeweb** [comment]
> Note that you most likely want to use a different tool like rsync to transfer files to the USB drive. Syncthing is not made for this use case and using it nevertheless seems quite suboptimal.

Thank you for your reply.

Well, so far I haven't had any issues using Syncthing to sync my laptop with my USB drive, and it's by far the only simple, fast, offline way I've used, including my cell phone.

I've tried other alternatives, but either they don't have a portable version, or they don't have phone support, so I would have to use two or three different applications.

Basically, SyncTrayzor is a kind of web viewer, because the configuration process is very minimal, everything else is Syncthing itself.

**Comment by Martchus** [comment]
Syncthing Tray can also be just that. You can completely ignore the built-in launcher and just let it connect to an existing Syncthing instance. The built-in web view is not supported by my Windows build but you can use the option to open the web-based UI in a Chromium-based browser to have something similar.

**Comment by gabeweb** [comment]
> Checkout the help text via `--help`. You'll find an option to spawn multiple instances. This should probably be mentioned in the README as well.
> 
> Parallel running instances will still use the same config by default. You can place the config file right next to the exe for a portable installation that uses its own config, e.g. on the USB drive. The README contains a section explaining where the config is located.

Hi, again.

After trying several times (and really, several times) to get Syncthing Tray to be configured, on the one hand, on my laptop, and on the other hand, on my USB drive on the same laptop, I was also able to get two copies of Syncthing Tray running at the same time on the same device.

But my question now is, does this have any relation to the "instances" configuration shown in Syncthing Tray, or is that something else entirely? At least, from the copy on my laptop, I created two instances, but I see that in the Startup, it remains the same, regardless of which instance is selected. In my copy on the USB drive, I didn't create any additional instances, however, its configuration and folders are maintained.

Note that I have simply run the Syncthing Tray application by double-clicking the executable and not from the command line. I haven't customized any commands either, such as `--single-instance` or anything like that; I simply added the following:

- `syncthingtray.ini` alongside the Syncthing Tray executable,
- the `config` folder inside the `Syncthing` folder, where the Syncthing executable is located.

Thanks for the help (even though I can be a bit stubborn sometimes).

**Comment by Martchus** [comment]
> But my question now is, does this have any relation to the "instances" configuration shown in Syncthing Tray, or is that something else entirely?

Actually, a good question. This perhaps also needs clarification on the README. The answer is that those are two different features. However, those different features are still a bit related.

* To help with a few use cases, Syncthing Tray behaves as a "single instance application". Here "instance" refers to the process of Syncthing Tray.
    * So by default only one Syncthing Tray process runs at a time. If you use the launcher that means there's also only one Syncthing process being launched at a time. If you start a 2nd one, the process that is already running handles showing a 2nd icon or other command-line options such as `--trigger`. The 2nd process itself terminates after passing the command to the already running process.
    * This has the advantage that there is always only one process and not multiple ones that would e.g. override each other's config. For options like `--trigger` it also wouldn't make much sense if a completely new process handles it.
    * I find this behavior useful and in most cases it is at least not disturbing. In case you want to start two different instances in parallel with distinct configurations it makes no sense of course. Then I'd recommend using `--new-instance`.
* Syncthing Tray supports configuring how to connect to Syncthing for more than just one Syncthing instance. Here "instance" relates to Syncthing itself and not the wrapper. The idea is that one can easily switch configurations to e.g. show the status of a server as needed instead of the status of the local Syncthing instance. I must say that I didn't find this feature super useful after all except for testing. However, there are some users actually using it.

You can combine these two features, e.g. `syncthingtray qt-widgets-gui --connection server` would spawn a 2nd tray icon for your server. Due to the single instance behavior there's still only one process running (which also makes sense in this case).

> I haven't customized any commands either, such as --single-instance or anything like that;

I guess you want `--new-instance` (as mentioned before) and *not* `--single-instance` which would still keep the mentioned single instance behavior.

> even though I can be a bit stubborn sometimes

No problem. Your use case simply something I haven't considered yet. As mentioned, I'll update the README when I have time to cover it. Perhaps it also makes sense to make it easier to get rid of the single instance behavior because having to use `--new-instance` is perhaps not very convenient. It would probably make most sense if a 2nd process would still lookup its config file and if it differs from the config file of the already launched process it will not terminate as if `--new-instance` was specified.

**Comment by gabeweb** [comment]
> I guess you want `--new-instance` (as mentioned before) and _not_ `--single-instance` which would still keep the mentioned single instance behavior.

Okay, so I had to make a shortcut (and tweak the auto-start settings) with the `--new-instance` command on my laptop, and I also messed with a shortcut for the copy on my USB stick (I use PortableApps to keep my portable stuff tidy), and the system tray icon for the same instance no longer appears repeated.

Sorry for bugging you, but is there any way to show the connection label (instance) without having to make another instance? I had to make two instances just so I could tell which window belongs to which copy (besides the system icon, which I also made different):

![Image](https://github.com/user-attachments/assets/de4bd7c0-a435-4e48-b945-f08153877726)

I'm really happy I got both copies working now.
